src/Messaging/Messages.ts

```diff
@@ -0,0 +1,24 @@
+import { isClaim, PoetBlockAnchor } from '@po.et/poet-js'
+import { pipe, not, allPass } from 'ramda'
+
+export interface BlockDownloaded {
+  readonly block: LightBlock
+  readonly poetBlockAnchors: ReadonlyArray<PoetBlockAnchor>
+}
+
+export interface LightBlock {
+  readonly hash: string
+  readonly previousHash: string
+  readonly height: number
+}
+
+export const isBlockDownloaded = (messageContent: any): messageContent is BlockDownloaded => {
+  const { poetBlockAnchors } = messageContent
+
+  const isNotClaim = pipe(
+    isClaim,
+    not
+  )
+
+  return allPass([Array.isArray(poetBlockAnchors), !poetBlockAnchors.some(isNotClaim)])
```

Comment:
allPass is supposed to take a list of predicates and have a value passed into the returned function

Reply:
Ugh you're right. Thanks.

Reply:
Your welcome just finished reviewing the rest of the PR, it looks good. Just `isBlockDownloaded` that needs some work, might be worth adding some unit tests for it?